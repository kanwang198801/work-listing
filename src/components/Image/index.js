import React, { memo } from 'react';
import PropTypes from 'prop-types';
import { Image } from 'antd';
import styles from './styles.module.css';
import placeholder from '../../assets/placeholder.jpg';

const WIDTH = 135;
const HEIGHT = 135;
const SingleImage = ({ work }) => {
  return (
    <div key={work.id} className={styles.image}>
      <Image
        width={WIDTH}
        height={HEIGHT}
        src={work.urls?.[0]?.link}
        alt={work.filename}
        preview={{ src: work.urls?.[2]?.link }}
        fallback={placeholder}
        placeholder={
          <Image
            preview={false}
            src={placeholder}
            width={WIDTH}
            height={HEIGHT}
          />
        }
      />
    </div>
  );
};

SingleImage.propTypes = {
  work: PropTypes.object.isRequired
};

export default memo(SingleImage);
