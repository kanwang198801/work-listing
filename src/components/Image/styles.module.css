.image {
  margin: 3px 8px;
  display: block;
}
