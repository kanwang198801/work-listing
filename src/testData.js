export const works = [
  {
    id: '31820',
    filename: '162042.jpg',
    imageWidth: '2206',
    imageHeight: '2205',
    urls: [
      {
        type: 'small',
        link: 'http://ih1.redbubble.net/work.31820.1.flat,135x135,075,f.jpg'
      },
      {
        type: 'medium',
        link: 'http://ih1.redbubble.net/work.31820.1.flat,300x300,075,f.jpg'
      },
      {
        type: 'large',
        link: 'http://ih1.redbubble.net/work.31820.1.flat,550x550,075,f.jpg'
      }
    ],
    exif: {
      model: 'NIKON D80',
      software: 'Adobe Photoshop CS3 Macintosh',
      isoSpeedRatings: '100',
      dateTime: 'Sun Aug 26 12:55:02 +1000 2007',
      make: 'NIKON CORPORATION'
    }
  },
  {
    id: '26583',
    filename: '27112.jpg',
    imageWidth: '1109',
    imageHeight: '1104',
    urls: [
      {
        type: 'small',
        link: 'http://ih1.redbubble.net/work.26583.1.flat,135x135,075,f.jpg'
      },
      {
        type: 'medium',
        link: 'http://ih1.redbubble.net/work.26583.1.flat,300x300,075,f.jpg'
      },
      {
        type: 'large',
        link: 'http://ih1.redbubble.net/work.26583.1.flat,550x550,075,f.jpg'
      }
    ],
    exif: {
      model: '',
      software: 'Adobe Photoshop CS2 Windows',
      isoSpeedRatings: '',
      dateTime: 'Fri Mar 30 19:11:10 +1000 2007',
      make: ''
    }
  }
];

export const filters = [
  { type: 'model', name: 'NIKON D80' },
  { type: 'make', name: 'NIKON CORPORATION' }
];
