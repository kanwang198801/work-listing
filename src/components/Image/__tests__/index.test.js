import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import Image from '../index';
import { works } from '../../../testData';

describe('Image Test Suite', () => {
  it('snapshot', () => {
    const tree = renderer.create(<Image work={works[0]} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the Image component', () => {
    const component = render(<Image work={works[0]} />);
    expect(component).toBeTruthy();
  });
});
