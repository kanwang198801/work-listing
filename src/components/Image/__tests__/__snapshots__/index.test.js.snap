// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image Test Suite snapshot 1`] = `
<div
  className="image"
>
  <div
    className="ant-image"
    onClick={[Function]}
    style={
      Object {
        "height": 135,
        "width": 135,
      }
    }
  >
    <img
      alt="162042.jpg"
      className="ant-image-img"
      onError={[Function]}
      onLoad={[Function]}
      src="http://ih1.redbubble.net/work.31820.1.flat,135x135,075,f.jpg"
      style={
        Object {
          "height": 135,
        }
      }
    />
    <div
      aria-hidden="true"
      className="ant-image-placeholder"
    >
      <div
        className="ant-image"
        style={
          Object {
            "height": 135,
            "width": 135,
          }
        }
      >
        <img
          className="ant-image-img"
          onError={[Function]}
          onLoad={[Function]}
          src="placeholder.jpg"
          style={
            Object {
              "height": 135,
            }
          }
        />
      </div>
    </div>
    <div
      className="ant-image-mask"
    >
      <div
        className="ant-image-mask-info"
      >
        <span
          aria-label="eye"
          className="anticon anticon-eye"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="eye"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M942.2 486.2C847.4 286.5 704.1 186 512 186c-192.2 0-335.4 100.5-430.2 300.3a60.3 60.3 0 000 51.5C176.6 737.5 319.9 838 512 838c192.2 0 335.4-100.5 430.2-300.3 7.7-16.2 7.7-35 0-51.5zM512 766c-161.3 0-279.4-81.8-362.7-254C232.6 339.8 350.7 258 512 258c161.3 0 279.4 81.8 362.7 254C791.5 684.2 673.4 766 512 766zm-4-430c-97.2 0-176 78.8-176 176s78.8 176 176 176 176-78.8 176-176-78.8-176-176-176zm0 288c-61.9 0-112-50.1-112-112s50.1-112 112-112 112 50.1 112 112-50.1 112-112 112z"
            />
          </svg>
        </span>
        Preview
      </div>
    </div>
  </div>
</div>
`;
